src/main/java/org/opentripplanner/routing/api/request/refactor/request/JourneyRequest.java

```diff
@@ -0,0 +1,47 @@
+package org.opentripplanner.routing.api.request.refactor.request;
+
+import org.opentripplanner.routing.core.TraverseMode;
+import org.opentripplanner.routing.core.TraverseModeSet;
+
+public class JourneyRequest {
+
+  // TODO: 2022-08-18 should it be here?
+  /**
+   * The set of TraverseModes allowed when doing creating sub requests and doing street routing. //
+   * TODO OTP2 Street routing requests should eventually be split into its own request class.
+   */
+  private TraverseModeSet streetSubRequestModes = new TraverseModeSet(TraverseMode.WALK);
+  private TransitRequest transit = new TransitRequest();
+  private StreetRequest access = new StreetRequest();
+  private StreetRequest egress = new StreetRequest();
+  private StreetRequest transfer = new StreetRequest();
+  private StreetRequest direct = new StreetRequest();
+
```

Comment:
There is little here to indicate that we run two routing searches, it might get more clear if we add to flags for it:
`boolean runTransitSearch`
`boolean runDirectSearch`
 Another way to do it, is the put the transit, access, egress and transfer into `TransitJourneyRequest` and use it here. Then the direct and transit is performed is present (not null). I like the second approach best, but you can decide.

Reply:
In the old the model there was no such fields but there obviously router was working anyway. Was there other way to determine whether we should run direct or transit search that I missed? If I add those two fields what's the proper place to add mapping to them and where should be used? Could navigate me little bit on which parts of the code I need to look at?